proxy_map.go

```diff
@@ -121,7 +114,111 @@ func (m *Map) NewLockContext(ctx context.Context) context.Context {
 	return context.WithValue(ctx, lockIDKey{}, lockID(m.refIDGen.NextID()))
 }
 
+type MapListener struct {
+	EntryAdded   func(event *EntryNotified)
+	EntryRemoved func(event *EntryNotified)
+	EntryUpdated func(event *EntryNotified)
+	EntryEvicted func(event *EntryNotified)
+	EntryExpired func(event *EntryNotified)
+	MapEvicted   func(event *EntryNotified)
+	MapCleared   func(event *EntryNotified)
+	EntryMerged  func(event *EntryNotified)
+	EntryLoaded  func(event *EntryNotified)
+}
+
+func (m *Map) prepareFlagsOfMapListener(listener MapListener) int32 {
+	var flags int32
+	if listener.EntryAdded != nil {
+		flagsSetOrClear(&flags, int32(EntryAdded), true)
+	}
+	if listener.EntryEvicted != nil {
+		flagsSetOrClear(&flags, int32(EntryEvicted), true)
+	}
+	if listener.EntryExpired != nil {
+		flagsSetOrClear(&flags, int32(EntryExpired), true)
+	}
+	if listener.EntryLoaded != nil {
+		flagsSetOrClear(&flags, int32(EntryLoaded), true)
+	}
+	if listener.EntryMerged != nil {
+		flagsSetOrClear(&flags, int32(EntryMerged), true)
+	}
+	if listener.EntryRemoved != nil {
+		flagsSetOrClear(&flags, int32(EntryRemoved), true)
+	}
+	if listener.EntryUpdated != nil {
+		flagsSetOrClear(&flags, int32(EntryUpdated), true)
+	}
+	if listener.MapCleared != nil {
+		flagsSetOrClear(&flags, int32(EntryAllCleared), true)
+	}
+	if listener.MapEvicted != nil {
+		flagsSetOrClear(&flags, int32(EntryAllEvicted), true)
+	}
+	return flags
+}
+
+func (m *Map) mapListenerEventHandler(listener MapListener) EntryNotifiedHandler {
+	return func(event *EntryNotified) {
+		switch event.EventType {
+		case EntryAdded:
+			listener.EntryAdded(event)
+			break
+		case EntryUpdated:
+			listener.EntryUpdated(event)
+			break
+		case EntryRemoved:
+			listener.EntryRemoved(event)
+			break
+		case EntryEvicted:
+			listener.EntryEvicted(event)
+			break
+		case EntryExpired:
+			listener.EntryExpired(event)
+			break
+		case EntryMerged:
+			listener.EntryMerged(event)
+			break
+		case EntryLoaded:
+			listener.EntryLoaded(event)
+			break
+		case EntryAllCleared:
+			listener.MapCleared(event)
+			break
+		case EntryAllEvicted:
+			listener.MapEvicted(event)
+			break
+		}
```

Comment:
Does it make sense to add a `default` branch here with a warning? Maybe something like:
```
m.logger.Warnf("unknown map event type: %d", event.EventType)
```